File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n]; 
        int total =  Arrays.stream(nums).sum();
        int left = 0;
        int right =total;
        for (int i =0; i < n; i++){
            ans[i] = i * nums[i] - left + right - (n - i) * nums[i];
            left += nums[i];
            right -= nums[i];
        }
        return ans;
    }
}

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n
        total = sum(nums)
        left = 0
        right = total
        for i in range(n):
            ans[i] = i * nums[i] - left + right - (n -i) * nums[i]
            left += nums[i]
            right -= nums[i]
        return ans